go/queue/queue.go

```diff
@@ -0,0 +1,108 @@
+// Package queue provides an interface to the work queue for the autograder instance.
+package queue
+
+import (
+	"github.com/streadway/amqp"
+)
+
+// Channel represents a connection to the queue service.
+type Channel struct {
+	*amqp.Connection
+	*amqp.Channel
+	queues          map[string]amqp.Queue
+	receiveChannels map[string]<-chan amqp.Delivery
+}
+
+// Open takes a string spec and opens a connection to the queue.
+// Example of connection spec: "amqp://localhost:5672/".
+func Open(spec string) (*Channel, error) {
+	conn, err := amqp.Dial(spec)
+	if err != nil {
+		return nil, err
+	}
+	ch, err := conn.Channel()
+	if err != nil {
+		conn.Close()
+		return nil, err
+	}
+	return &Channel{
+		Connection: conn,
+		Channel:    ch,
+		queues:     make(map[string]amqp.Queue),
+	}, nil
+}
+
+func (ch *Channel) Close() error {
+	err1 := ch.Channel.Close()
+	err2 := ch.Connection.Close()
+	if err1 != nil {
+		return err1
+	}
+	return err2
+}
+
+func (ch *Channel) getQueue(queueName string) (amqp.Queue, error) {
+	q, ok := ch.queues[queueName]
+	if !ok {
```

Comment:
early-return to reduce the nest-level.

Maybe, you can write

```golang
if q, ok := ..; ok {
  return q, nil
}
```

Reply:
Done.